stan/math/prim/scal.hpp

```diff
@@ -21,7 +21,6 @@
 #include <stan/math/prim/scal/err/check_positive_finite.hpp>
 #include <stan/math/prim/scal/err/check_size_match.hpp>
 #include <stan/math/prim/scal/err/domain_error.hpp>
```

Comment:
Remove this include of `domain_error` (the include for `throw_domain_error` was already included).

Reply:
I had first removed it, then added this back to ease the transition for Stan. Are you sure I should remove it? I'll wait for your answer here before I push my changes.

Reply:
ahh... you're absolutely right. That would make things not break. 

To follow your logic, can you add back the include for `stan/math/prim/scal/err/domain_error_vec.hpp`? Also, could we add a `@deprecated use <code>thrrow_domain_error</code>` to the doc in `domain_error.hpp` and the appropriate doc to `domain_error_vec.hpp`?

Reply:
Cool, I've restored `domain_error_vec.hpp` with `@deprecated` tags (I had already added them to `domain_vector.hpp`, so I think I've covered all you requested.